lib/identity/realm.js

```diff
@@ -354,18 +418,32 @@ class Realm extends CryptoConsumer {
     )
     const tokenInfo = await credentialedDecodeResponse(finalRequest)
     const token = tokenInfo.access_token
-    const storedCreds = await credentialNoteCall(
-      this.storage.readNoteByName(
-        noteName,
-        cryptoKeyPair,
-        signingKeyPair,
-        {},
-        {
+    let storedCreds
+    if (federated) {
+      const keyNoteName = await this.crypto.hash(
+        `federated:${username}@realm:${this.realmName}`
```

Comment:
Ignore me if this is completely off, but would a user ever want to linked to multiple external idms? Do we need additional context in this naming scheme for later?

Reply:
so... somewhat no, that's not how Keycloak functions.. A user has a single federation link. _However_ this one note will work for any provider where Keycloak becomes the primary authentication factor. So if you were connected to two different LDAPs, or something, authenticating with either one would be enough to allow Keycloak to issue your session token, which could be used to request this broker key note from the hosted broker. Getting a valid session token proves you've authenticated with Keycloak, which in a federated world, is enough for the broker to release the credentials. It is important that in non-federated accounts this doesn't happen.